internal/venice-common/src/main/java/com/linkedin/venice/kafka/validation/checksum/AdHash.java

```diff
@@ -0,0 +1,73 @@
+package com.linkedin.venice.kafka.validation.checksum;
+
+import com.linkedin.venice.utils.ByteUtils;
+import java.util.zip.CRC32;
+
+
+/**
+ * This class AdHash is used to keep track of the checksum of the data incrementally.
+ * It adopts the idea of AdHash that:
+ * " A message x' which I want to hash may be a simple modification of a message x which I previously
+ *   hashed. If I have already computed the hash f(x) of x then, rather than re-computing f(x') from
+ *   scratch, I would like to just quickly 'update' the old hash value f(x) to the new value f(x').
+ *   An incremental hash function is one that permits this. "
+ *
+ * See the paper for details:
+ *   A New Paradigm for Collision-free Hashing: Incrementality at Reduced Cost.
+ *   (https://cseweb.ucsd.edu/~daniele/papers/IncHash.pdf)
+ *
+ * Its adoption in ZooKeeper:
+ *   Verify, And Then Trust: Data Inconsistency Detection in ZooKeeper.
+ *   (https://dl.acm.org/doi/abs/10.1145/3578358.3591328)
+ */
+
+public class AdHash extends CheckSum {
+  private volatile long incrementalDigest;
```

Comment:
Does it need to be volatile?

Reply:
We can read this value in a different thread e.g. SIT thread. Please check `StoreIngestionTask.checkLongRunningTaskState`

Reply:
In `SIT::cloneProducerStates`?

Reply:
correct.